tests/unit/test_start_handlers.py

```diff
@@ -48,6 +48,40 @@ async def test_get_index_ni(self):
         self.assertEqual(contents.count('input--text'), 1)
         self.assertIn('type="submit"', contents)
 
+    @unittest_run_loop
+    async def test_post_index_en_with_retry_503(self):
+        with aioresponses(passthrough=[str(self.server._root)]) as mocked:
+            mocked.get(self.rhsvc_url, status=503)
+            mocked.get(self.rhsvc_url, status=503)
+            mocked.get(self.rhsvc_url, payload=self.uac_json_en)
+
+            response = await self.client.request('POST',
+                                                 self.post_start_en,
+                                                 allow_redirects=False,
+                                                 data=self.start_data_valid)
+
+        self.assertEqual(response.status, 302)
+        self.assertIn('/start/confirm-address',
+                      response.headers['Location'])
+
+    @unittest_run_loop
+    async def test_post_index_en_with_retry_ConnectionError(self):
+        with aioresponses(passthrough=[str(self.server._root)]) as mocked:
+            mocked.get(self.rhsvc_url,
```

Comment:
Duplication here as well

Reply:
I needed to multiple mocks so that each retry returns a 503 the number of times it goes round the loop. However, I have recoded this to hopefully be more explicit.